packages/api/src/command/medical/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -0,0 +1,99 @@
+import { XCA_DQ_STRING, XCA_DR_STRING, XCPD_STRING } from "@metriport/carequality-sdk/common/util";
+import { Address } from "@metriport/carequality-sdk/models/address";
+import { Contained } from "@metriport/carequality-sdk/models/contained";
+import { Organization } from "@metriport/carequality-sdk/models/organization";
+import { normalizeOid } from "@metriport/shared";
+import { CQDirectoryEntryData } from "../../../domain/medical/cq-directory";
+
+export type XCUrls = {
+  urlXCPD: string;
+  urlDQ?: string;
+  urlDR?: string;
+};
+
+export function parseCQDirectoryEntries(orgsInput: Organization[]): CQDirectoryEntryData[] {
+  const orgs = orgsInput.flatMap(org => {
+    const orgOid = org?.identifier?.value?.value;
+    if (!orgOid) return [];
+
+    const url = getUrls(org?.contained);
+    if (!url?.urlXCPD) return [];
+    let oid = org?.identifier?.value?.value;
+    try {
+      oid = normalizeOid(org?.identifier?.value?.value);
+    } catch (err) {
+      console.log(`Organization ${org.name?.value} has invalid OID: ${oid}`);
+      return [];
+    }
+    const coordinates = getCoordinates(org?.address);
+
+    const state = getState(org.address);
+    const orgData: CQDirectoryEntryData = {
+      oid,
+      name: org.name?.value ?? undefined,
+      urlXCPD: url.urlXCPD,
+      urlDQ: url.urlDQ,
+      urlDR: url.urlDR,
+      lat: coordinates?.lat ?? undefined,
+      lon: coordinates?.lon ?? undefined,
+      data: {
+        ...org,
+      },
+      state,
+    };
+    return orgData;
+  });
+  return orgs;
+}
+
+function getUrls(contained: Contained): XCUrls | undefined {
+  const endpointMap: Record<string, string> = {};
+
+  contained?.forEach(c => {
+    const ext = c?.Endpoint.extension.extension.find(ext => ext.url === "Transaction");
```

Comment:
Is there a type for `ext.url` that makes `Transaction` validated by the compiler, or is it `any`? If `any`, I'd recommend we have `Transaction` as a shared const somewhere

Reply:
yea, it's just a string right now, so adding a const sounds like a good idea